core/src/main/java/io/parsingdata/metal/token/Until.java

```diff
@@ -101,8 +101,8 @@ public Until(final String name, final ValueExpression initialSize, final ValueEx
 
     private Trampoline<Optional<ParseState>> iterate(final Environment environment, final BigInteger currentSize, final BigInteger stepSize, final BigInteger maxSize) {
         if (stepSize.compareTo(ZERO) == 0 ||
-            stepSize.compareTo(ZERO) > 0 && currentSize.compareTo(maxSize) > 0 ||
-            stepSize.compareTo(ZERO) < 0 && currentSize.compareTo(maxSize) < 0) {
+            (stepSize.compareTo(ZERO) > 0 && currentSize.compareTo(maxSize) > 0) ||
+            (stepSize.compareTo(ZERO) < 0 && currentSize.compareTo(maxSize) < 0)) {
```

Comment:
We have to say "we told you so", but... :stuck_out_tongue_winking_eye: :blush: :+1: